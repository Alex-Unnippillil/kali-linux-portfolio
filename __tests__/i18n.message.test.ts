import {
  FALLBACK_LOCALE,
  createMessageFormatter,
  defaultCatalogs,
  getMessageFormatter,
} from '../i18n/message';

describe('ICU message formatting', () => {
  it('formats pluralised trash day strings across locales', () => {
    const counts = [0, 1, 2, 5];
    const locales = ['en', 'es', 'fr'];
    const formatted = Object.fromEntries(
      locales.map((locale) => {
        const formatter = createMessageFormatter({
          locale,
          fallbackLocale: FALLBACK_LOCALE,
          catalogs: defaultCatalogs,
        });
        return [
          locale,
          counts.map((count) =>
            formatter.format('trash.daysRemaining', { count }),
          ),
        ];
      }),
    );

    expect(formatted).toMatchSnapshot();
  });

  it('supports gender aware selections', () => {
    const english = getMessageFormatter('en');
    const spanish = getMessageFormatter('es');

    const samples = {
      enMale: english.format('trash.genderedActor', {
        gender: 'male',
        name: 'Alex',
        count: 2,
      }),
      enFemale: english.format('trash.genderedActor', {
        gender: 'female',
        name: 'Riley',
        count: 1,
      }),
      enNeutral: english.format('trash.genderedActor', {
        gender: 'other',
        name: 'Sam',
        count: 3,
      }),
      esNeutral: spanish.format('trash.genderedActor', {
        gender: 'other',
        name: 'Mar',
        count: 4,
      }),
    };

    expect(samples).toMatchSnapshot();
  });

  it('falls back to the default locale for missing translations', () => {
    const formatter = getMessageFormatter('fr');
    expect(formatter.resolvedLocale).toBe('en');
    expect(
      formatter.format('youtube.playlistCount', { count: 3 })
    ).toBe('3 playlists');
  });
});
