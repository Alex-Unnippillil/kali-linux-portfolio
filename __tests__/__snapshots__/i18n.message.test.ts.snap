// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ICU message formatting formats pluralised trash day strings across locales 1`] = `
{
  "en": [
    "0 days left",
    "1 day left",
    "2 days left",
    "5 days left",
  ],
  "es": [
    "Quedan 0 días",
    "Queda 1 día",
    "Quedan 2 días",
    "Quedan 5 días",
  ],
  "fr": [
    "0 days left",
    "1 day left",
    "2 days left",
    "5 days left",
  ],
}
`;

exports[`ICU message formatting supports gender aware selections 1`] = `
{
  "enFemale": "Riley deleted her file",
  "enMale": "Alex deleted his files",
  "enNeutral": "Sam deleted their files",
  "esNeutral": "Mar eliminó sus archivos",
}
`;
